pytest-selenium/tests/test_search.py

```diff
@@ -0,0 +1,74 @@
+from random import choice
+from string import digits, ascii_letters
+
+from pages.content import Content
+from tests import markers
+
+
+@markers.test_case("C543235")
+@markers.parametrize("page_slug", ["preface"])
+@markers.nondestructive
+def test_message_when_search_yields_no_results(selenium, base_url, book_slug, page_slug):
+    # GIVEN: Book page is loaded
+    content = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+    toolbar = content.toolbar
+    mobile = content.mobile
+    search_sidebar = content.sidebar.search_sidebar
+    page_url_before_search = content.current_url
+
+    # Using regex to create a random search term
+    search_term = "".join(choice(digits + ascii_letters) for i in range(25))
+
+    # WHEN: Search is triggered for a term which yields no results
+    if content.is_desktop:
+        toolbar.search_for(search_term)
+
+    if content.is_mobile:
+        mobile.search_for(search_term)
+
+    # THEN: Search sidebar displays no results message
+    assert (
+        search_sidebar.no_results_message == "Sorry, no results found for   ‘"
+        "{}"
+        "’".format(search_term)
+    )
+
+    # AND: User stays in the same page as before executing the search
+    assert content.current_url == page_url_before_search
+
+    # AND: Scroll position of the content is not changed during the search
+    if content.is_mobile:
+        toolbar.click_search_icon()
+
+    assert content.page_not_scrolled
+
+
+@markers.test_case("C568506")
+@markers.parametrize("page_slug", ["preface"])
+@markers.nondestructive
+def test_scroll_position_when_search_yields_no_results(selenium, base_url, book_slug, page_slug):
+    # GIVEN: Book page is loaded
+    content = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+    toolbar = content.toolbar
+    mobile = content.mobile
+
+    # Using regex to create a random search term
+    search_term = "".join(choice(digits + ascii_letters) for i in range(25))
+
+    # WHEN: Scroll the page down
+    content.scroll_through_page()
+
+    scroll_position_before_search = content.scroll_position
+
+    # AND: Search is triggered for a term which yields no results
+    if content.is_desktop:
+        toolbar.search_for(search_term)
+
+    if content.is_mobile:
+        mobile.search_for(search_term)
+        toolbar.click_search_icon()
```

Comment:
shouldn't `search_for`'s call to `trigger_search` make this unnecessary?

Reply:
`search_for` loads the search sidebar. But I want to come back to the content (or close the search sidebar in mobile) for which I need to click the search icon again.

May be I am doing it wrong or I may need to add more comments to make the test clear. For now, I am adding more comments in the test and you let me know if it makes sense